docs/source/developer_documentation/reports.md

```diff
@@ -0,0 +1,83 @@
+# Technical Report Documentation
+
+## Creating reports
+### Location of the report code
+The code of the reports is located inside the `rocky/reports/report_types` folder.
+
+### Steps to create a new report
+Make sure Rocky is running so you can see the changes you are about to make.
+
+1.	Navigate to the `report_types` folder that contains all the different report types.
+2.	Create a new folder with a descriptive name ending with `_report`. Alternatively, you can duplicate an existing folder, modify the names, and clear its contents. (If you’ve done this, continue to step 3.)
+3.	Inside the new folder, create the following files: `__init__.py`, `report.html`, and `report.py`.
+4.	Define a class within `report.py` with the name of your report and include the following variables:
+```
+class YourNameReport(Report):
+    # The id of your report:
+    id = "your_report_name"
+    # The name users will see:
+    name = _("Your Report Name")
+    description = _("Give a description to your new report.")
+    # All the required and optional plugins (can be empty lists):
+    plugins = {"required": ["nmap"], "optional": ["shodan", "nmap-udp", "nmap-ports", "nmap-ip-range"]}
+    # The OOI types that can serve as input to generate this report:
+    input_ooi_types = {Hostname, IPAddressV4, IPAddressV6}
+    # Path of report.html:
+    template_path = "your_report_name/report.html"
+```
+5.	Open `reports/report_types/helpers.py` and add your new class to the `REPORTS` constant list.
+6.	Implement a method within `report.py` to gather the required data for report generation. See the [Collecting data](#collecting-data) section for more information.
+7.	Design the HTML structure for your report within `report.html`. The generated data from `report.py` can be used with Django Template in this file. For example by referring to the returned value like `{{ data }}`.
+8.	Save your changes and refresh the page to see the changes you made immediately.
+
+### Collecting data
+There are two ways to collect data:
+- Create a method called `collect_data` and use the `self.octopoes_api_connector.query_many()` method to data for multiple OOIs at the same time. This can reduce the number of queries to Octopoes, which is better for report performance.
+- Create a method called `generate_data`. This method is the old way of receiving the data from Octopoes. It is less efficient and therefore should only be used with `self.octopoes_api_connector.get_tree()` to collect data from a tree.
```

Comment:
I don't think we should keep two ways to do this, the two ways only exist because some reports still need to be changed to `collect_data`. `get_tree` can be used in `collect_data`, it is just less efficient because we need to do call for every OOI.

```suggestion
Data collection happens in the `collect_data` method that the report class should implement. Using the `self.octopoes_api_connector.query_many()` method data for multiple OOIs can be requested in one call. This is better for performance and is the preferred way to fetch data. Methods such as  `self.octopoes_api_connector.get_tree()` that work on a single OOI should only used if the multiple OOI methods don't provide the data that is needed.

The `generate_data` method only works on single OOIs and should not be used in new reports. As soon as the existing reports that implement `generate_data` have been moved over to `collect_data` suppor for `generate_data` will be removed.
```